File: leetcode/C++/best-time-to-buy-and-sell-stock-ii.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int sol = 0, s=-prices[0], b=0;
        for(int i=1; i<prices.size(); ++i){
            b=max(b, s+prices[i]);
            s=max(s, b-prices[i]);
            sol=max(sol,b);
        }
        return sol;
    }
};

File: leetcode/Python/best-time-to-buy-and-sell-stock-ii.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        sol, s, b = 0, -prices[0], 0
        for i in range(1,len(prices)):
            s, b = max(s, b-prices[i]), max(b, s+prices[i])
            sol=max(sol,b)
        return sol